utils/build-script-impl

```diff
@@ -3384,6 +3385,49 @@ for host in "${ALL_HOSTS[@]}"; do
 done
 # Everything is 'installed', but some products may be awaiting lipo.
 
+function build_symlink_darwin_toolchain() {
+    local host="$1"
+    local host_install_destdir="$(get_host_install_destdir ${host})"
+
+    echo "--- Creating xctoolchain with symlinks ---"
+    echo "-- Create Info.plist --"
+    # FIXME: Duplicated from build_and_test_installable_package
+    PLISTBUDDY_BIN="/usr/libexec/PlistBuddy"
+
+    DARWIN_TOOLCHAIN_INFO_PLIST="${host_install_destdir}${TOOLCHAIN_PREFIX}/Info.plist"
+    COMPATIBILITY_VERSION=2
+    COMPATIBILITY_VERSION_DISPLAY_STRING="Xcode 8.0"
+    DARWIN_TOOLCHAIN_CREATED_DATE="$(date -u +'%a %b %d %T GMT %Y')"
+
+    echo "-- Removing: ${DARWIN_TOOLCHAIN_INFO_PLIST}"
+    call rm -f ${DARWIN_TOOLCHAIN_INFO_PLIST}
+
+    call ${PLISTBUDDY_BIN} -c "Add DisplayName string '${DARWIN_TOOLCHAIN_DISPLAY_NAME}'" "${DARWIN_TOOLCHAIN_INFO_PLIST}"
+    call ${PLISTBUDDY_BIN} -c "Add ShortDisplayName string '${DARWIN_TOOLCHAIN_DISPLAY_NAME_SHORT}'" "${DARWIN_TOOLCHAIN_INFO_PLIST}"
+    call ${PLISTBUDDY_BIN} -c "Add CreatedDate date '${DARWIN_TOOLCHAIN_CREATED_DATE}'" "${DARWIN_TOOLCHAIN_INFO_PLIST}"
+    call ${PLISTBUDDY_BIN} -c "Add CompatibilityVersion integer ${COMPATIBILITY_VERSION}" "${DARWIN_TOOLCHAIN_INFO_PLIST}"
+    call ${PLISTBUDDY_BIN} -c "Add CompatibilityVersionDisplayString string ${COMPATIBILITY_VERSION_DISPLAY_STRING}" "${DARWIN_TOOLCHAIN_INFO_PLIST}"
+    call ${PLISTBUDDY_BIN} -c "Add Version string '${DARWIN_TOOLCHAIN_VERSION}'" "${DARWIN_TOOLCHAIN_INFO_PLIST}"
+    call ${PLISTBUDDY_BIN} -c "Add CFBundleIdentifier string '${DARWIN_TOOLCHAIN_BUNDLE_IDENTIFIER}'" "${DARWIN_TOOLCHAIN_INFO_PLIST}"
+    call ${PLISTBUDDY_BIN} -c "Add OverrideBuildSettings dict" "${DARWIN_TOOLCHAIN_INFO_PLIST}"
+    call ${PLISTBUDDY_BIN} -c "Add OverrideBuildSettings:ENABLE_BITCODE string 'NO'" "${DARWIN_TOOLCHAIN_INFO_PLIST}"
+    call ${PLISTBUDDY_BIN} -c "Add OverrideBuildSettings:SWIFT_DISABLE_REQUIRED_ARCLITE string 'YES'" "${DARWIN_TOOLCHAIN_INFO_PLIST}"
+    call ${PLISTBUDDY_BIN} -c "Add OverrideBuildSettings:SWIFT_LINK_OBJC_RUNTIME string 'YES'" "${DARWIN_TOOLCHAIN_INFO_PLIST}"
+
+    call mkdir -p "${host_install_destdir}${TOOLCHAIN_PREFIX}/usr/bin"
+    call ln -sf "$(build_directory_bin ${host} swift)/bin/"{swift,swiftc,swift-demangle,swift-stdlib-tool} "${host_install_destdir}${TOOLCHAIN_PREFIX}/usr/bin"
```

Comment:
When I run this locally, I seem to be getting the path `/Users/bgesiak/Source/apple/build/Ninja-ReleaseAssert/swift-macosx-x86_64/bin/bin/swiftc` -- note the extra `/bin`.  The following works for me:

```diff
diff --git a/utils/build-script-impl b/utils/build-script-impl
index 512aeff8f4..4a0e6e6a28 100755
--- a/utils/build-script-impl
+++ b/utils/build-script-impl
@@ -3415,7 +3415,7 @@ function build_symlink_darwin_toolchain() {
     call ${PLISTBUDDY_BIN} -c "Add OverrideBuildSettings:SWIFT_LINK_OBJC_RUNTIME string 'YES'" "${DARWIN_TOOLCHAIN_INFO_PLIST}"

     call mkdir -p "${host_install_destdir}${TOOLCHAIN_PREFIX}/usr/bin"
-    call ln -sf "$(build_directory_bin ${host} swift)/bin/"{swift,swiftc,swift-demangle,swift-stdlib-tool} "${host_install_destdir}${TOOLCHAIN_PREFIX}/usr/bin"
+    call ln -sf "$(build_directory_bin ${host} swift)/"{swift,swiftc,swift-demangle,swift-stdlib-tool} "${host_install_destdir}${TOOLCHAIN_PREFIX}/usr/bin"
     call mkdir -p "${host_install_destdir}${TOOLCHAIN_PREFIX}/usr/lib"
     call ln -sf "$(build_directory_bin ${host} swift)/../lib/"{swift,swift_static,libswiftDemangle.dylib,sourcekitd.framework} "${host_install_destdir}${TOOLCHAIN_PREFIX}/usr/lib"
```

Reply:
Whoops, I fixed this for LLDB but not for Swift.